### PR TITLE
Adding basic deploy to heroku orb.

### PR DESCRIPTION
**Pull request by bradylill**
- Simplified heroku cli install
- Added basic deploy using an API set in env variables

Working example here: https://github.com/bradylill/heroku-orb-test

### CODE REVIEW

File: src/heroku/orb.yml [hunk 1]
```diff
@@ -1,35 +1,30 @@
...
             else
               echo "Heroku is already installed. No operation was performed."
             fi
+
+  deploy:
+    steps:
+      - run:
+          name: Deploy Master to Heroku
+          command: |
+            git push https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP_NAME.git master
```
Comment (ndintenfass) [inline src/heroku/orb.yml]: We should make the branch name here a parameter that can default to master (or be required )

File: .circleci/config.yml [hunk 2]
```diff
@@ -13,7 +13,8 @@ jobs:
       - image: singapore/lint-condo
     steps:
       - checkout
-      - run: yamllint -d relaxed .
+      - run:
+          'yamllint -d "{extends: relaxed, rules: {line-length: {max: 200, allow-non-breakable-inline-mappings: true}}}" .'
```
Comment (marcomorain) [inline .circleci/config.yml]: Can you put these in a .yamllint file please?

File: src/heroku/orb.yml [hunk 3]
```diff
@@ -1,35 +1,67 @@
...
+      app-name:
+        description:
+          "The name of your Heroku App.
+          For backwards compatibility the literal value
+          `$HEROKU_APP_NAME` is the default, so you can
+          easily use this command by setting an environment
+          variable called HEROKU_APP_NAME"
+        type: string
+        default: $HEROKU_APP_NAME
+      api-key:
```
Comment (marcomorain) [inline src/heroku/orb.yml]: We should remove this IMO - secrets should be set in env vars, not in config.

File: src/heroku/orb.yml [hunk 4]
```diff
@@ -1,35 +1,67 @@
...
+        description:
+          "The API key to use. defaulting to the literal value $HEROKU_API_KEY, so you can
+          populate the environment variable HEROKU_API_KEY and not need to declare a value in code."
+        type: string
+        default: $HEROKU_API_KEY
+      branch:
```
Comment (marcomorain) [inline src/heroku/orb.yml]: Branch filtering should be done at the workflow definition level - let’s remove branch filtering from the job.

File: src/heroku/orb.yml (hunk 3 above)
Comment (ndintenfass) [reply src/heroku/orb.yml]: This does that, but creating an opaque, hard-coded dependency on what those variables can be called would make this harder to use and more prone to frustration. Passing the env vars as the values gives both.

File: src/heroku/orb.yml (hunk 4 above)
Comment (ndintenfass) [reply src/heroku/orb.yml]: In this case the primary audience is people moving from 1.0 or onboarding very basic apps. This way they can choose to put a filter here in a single line without needing to understand workflows at first. The default behavior is a noop, so this is innocuous.

File: .yamllint [hunk 5]
```diff
@@ -0,0 +1,6 @@
+extends: relaxed
+
+rules: 
+    line-length: 
+        max: 200 
+        allow-non-breakable-inline-mappings: true
```
Comment (ndintenfass) [inline .yamllint]: Ironically, this lack of newline would not pass the linter if this were linted.

Comment (bradylill) [reply .yamllint]: That's what nathanlint is for! XD

File: src/heroku/orb.yml [hunk 6]
```diff
@@ -1,35 +1,109 @@
...
+      - run:
+          name: Deploy branch to Heroku via git push
+          command: |
+            if [ "<< paramaters.only-branch >>" == "" || "${CIRCLE_BRANCH}" == "<< paramaters.only-branch >>" ]; then
+              git push https://heroku:<< parameters.api-key >>@git.heroku.com/<< parameters.app-name >>.git << parameters.branch >>
+            fi
+jobs:
+  deploy:
+    parameters:
+      app-name:
```
Comment (ndintenfass) [inline src/heroku/orb.yml]: Not a change, but an observation - this pattern of repeating params across a job and a command that job uses is going to be fairly common, and creates a lot of redundancy. Might be nice to think of a way to create parameter groups that can be reused.

**Comment by ndintenfass** [comment]
lgtm

File: src/heroku/orb.yml (hunk 6 above)
Comment (bradylill) [reply src/heroku/orb.yml]: Was thinking the same thing while writing this...

**Comment by ndintenfass** [comment]
@marcomorain I'm going to merge this - if you still object to the final state we can make changes.

File: src/heroku/orb.yml (hunk 3 above)
Comment (marcomorain) [reply src/heroku/orb.yml]: Giving users a en empty spot in a config file where we expect them to put their API key will cause them to put their API key into config files. This is a very bad precent to set.

**Pull request merged**